### ISSUE TITLE
ValueTask defined in xunit.runner.visualstudio v3 causing conflict

### ISSUE THREAD

**Issue by 6bee** [xunit/visualstudio.xunit#431]
### Issue
Type conflict caused by `ValueTask<>` causes dynamic type resolution to fail.
The following two type definitions collide for tests executed on .NET Framework:
```
System.Threading.Tasks.ValueTask`1, System.Threading.Tasks.Extensions, Version=4.2.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
System.Threading.Tasks.ValueTask`1, xunit.runner.visualstudio.testadapter, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null
```

### Potential Resolution
Would it be possible to used _System.Threading.Tasks.Extensions_ package rather than defining your own `ValueTask<>`?

### Background
_[aqua-core](https://www.nuget.org/packages/aqua-core)_ is a library that allows to serialize type information to be carried and re-created across application boundaries, different .net runtime and platform versions. Types defined in multiple locations (loaded into the default loading context) do contradict the concept and cannot be included in corresponding models. Hence, we're currently blocked from updating to _xunit.runner.visualstudio.3.0.0_.

**Comment by bradwilson** [comment]
The reason you're seeing this is because the test adapter uses ILRepack so there is a private copy that's been repacked inside.

However, the more important question is: Why are you linking against `xunit.runner.visualstudio.testadapter`? Nobody should be doing that.

**Comment by 6bee** [comment]
There is a package reference to _xunit.runner.visualstudio_ for our unit test projects. No direct reference to _xunit.runner.visualstudio.testadapter_. 
However, I suspect since we're using reflection to resolve types by full-name (not assembly-qualified-name), looking-up all loaded assemblies from AppDomain.CurrentDomain.GetAssemblies(), we end-up using `System.Threading.Tasks.ValueTask<>` from _xunit.runner.visualstudio.testadapter_ assembly.

**Comment by bradwilson** [comment]
I'll look into this and see if there is anything we can do.

**Comment by bradwilson** [comment]
The issue is that we are forced by VSTest to copy all of our DLLs into your output folder. This is done  outside the build process where, if we shared any dependencies, they could resolve those dependencies for us (because you're not actually linking against our runner DLL).

This is why we use ILRepack in the first place: any dependencies that we have need to become internalized or else we have a potential conflict problem (your version vs. our version). Additionally, ILRepack normally marks everything that's imported as internal instead of public, but it seems we have exposed some things publicly which consume/produce `ValueTask<>`, so it has left them public:

![image](https://github.com/user-attachments/assets/faec6658-39bb-42f5-a258-7a081a44e989)

The visual difference here in ILSpy is subtle, but the types/namespaces which are public are black vs. the ones that are internal being gray.

I'll see if I can make some changes what's left as public so that our version of `ValueTask<>` shows up as internal rather than public.

**Comment by bradwilson** [comment]
Fix available in `3.0.1-pre.2` https://xunit.net/docs/using-ci-builds

I was able to reduce the public surface to just the one class that VSTest needs to find publicly, and the interfaces they define don't include `ValueTask<>`, everything should be good to go now. 👍🏼 

![image](https://github.com/user-attachments/assets/c926ad25-ad66-4121-b5c8-4b2035f1b3cb)

**Comment by 6bee** [comment]
This resolves our issue since we now can prioritize based on visibility. 

Thank you so much @bradwilson for your responsiveness and the fix in such a short time. It is highly appreciated! ❤️
